qa/L0_config_search/test_config_generator.py

```diff
@@ -127,16 +141,30 @@ def generate_max_limit_with_param_and_model_config(self):
                 } for model in self.profile_models
             },
         }
-        self._write_file(3, 1, 2, 1, model_config)
+        total_param_count = self._calculate_total_params(
+            concurrency_count, instance_count)
+        self._write_file(total_param_count, 1, 2, 1, model_config)
 
     def generate_max_limit_with_dynamic_batch_disable(self):
+        concurrency_count = 2
+        instance_count = 2
         model_config = {
-            "run_config_search_max_concurrency": 2,
-            "run_config_search_max_instance_count": 2,
-            "run_config_search_preferred_batch_size_disable": True,
+            "run_config_search_max_concurrency": concurrency_count,
+            "run_config_search_max_instance_count": instance_count,
             "profile_models": self.profile_models,
         }
-        self._write_file(6, 2, 4, 1, model_config)
+        total_param_count = self._calculate_total_params(
+            concurrency_count, instance_count)
+        self._write_file(total_param_count, 2, 4, 1, model_config)
+
+    def _calculate_total_params(self,
+                                concurrency_count,
+                                instance_count,
+                                default_config_count=1):
+        """
+        Calculate the total number of configs that MA will try
```

Comment:
Some more details of the args would be nice:

Given the count of concurrencies and instances to sweep over, calculate and return the total number of parameter combinations that MA will try.  default_config_count indicates how many extra model configs will be run in addition to the normal sweep, and should be set to 0 if the default config is already part of the sweep space.